scripts/sct_dmri_moco.py

```diff
@@ -40,78 +41,109 @@ def get_parser():
+            '    -group-wise (-g)\n'
+            '    -slice-wise regularized along z using polynomial function (-param). For more info about the method, '
+            'type: isct_antsSliceRegularizedRegistration\n'
+            '    -masking (-m)\n'
+            '    -iterative averaging of target volume\n'
+        ),
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help="Diffusion data. Example: dmri.nii.gz"
+    )
+    mandatory.add_argument(
+        '-bvec',
+        metavar=Metavar.file,
+        required=True,
+        help='Bvecs file. Example: bvecs.txt'
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        '-bval',
+        metavar=Metavar.file,
+        default=param_default.fname_bvals,
+        help='Bvals file. Example: bvals.nii.gz',
+    )
+    optional.add_argument(
+        '-bvalmin',
+        type=float,
+        metavar=Metavar.float,
+        default=param_default.bval_min,
+        help='B-value threshold (in s/mm2) below which data is considered as b=0. Example: 50.0',
+    )
+    optional.add_argument(
+        '-g',
+        type=int,
+        metavar=Metavar.int,
+        default=param_default.group_size,
+        help='Group nvols successive dMRI volumes for more robustness. Example: 2',
+    )
+    optional.add_argument(
+        '-m',
+        metavar=Metavar.file,
+        default=param_default.fname_mask,
+        help='Binary mask to limit voxels considered by the registration metric. Example: dmri_mask.nii.gz',
+    )
+    optional.add_argument(
+        '-param',
+        metavar=Metavar.list,
+        type=list_type(',', str),
+        help=f"R|Advanced parameters. Assign value with \"=\", and separate arguments with \",\".\n"
+             f"    -poly [int]: Degree of polynomial function used for regularization along Z. For no regularization set to "
+             f"0. Default={param_default.poly}.\n"
+             f"    -smooth [mm]: Smoothing kernel. Default={param_default.smooth}.\n"
+             f"    -metric {{MI, MeanSquares, CC}}: Metric used for registration. Default={param_default.metric}.\n"
+             f"    -gradStep [float]: Searching step used by registration algorithm. The higher the more deformation "
+             f"allowed. Default={param_default.gradStep}.\n"
+             f"    -sample [None or 0-1]: Sampling rate used for registration metric. Default={param_default.sampling}.\n"
+    )
+    optional.add_argument(
+        '-x',
+        metavar=Metavar.str,
+        choices=['nn', 'linear', 'spline'],
```

Comment:
the choices are presently not visible from the usage. I _think_ this is because you set the metavar variable. See e.g. [sct_apply_transfo](https://github.com/neuropoly/spinalcordtoolbox/blob/master/scripts/sct_apply_transfo.py#L101), where this variable is not set (and the different choices are visible when launching the CLI's help)